### ISSUE TITLE
%timespan% before {date} causes {date} to lose %timespan%

### ISSUE THREAD

**Issue by not-3than** [SkriptLang/Skript#4509]
### Skript/Server Version

```
[19:46:29 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[19:46:29 INFO]: [Skript] Skript's documentation can be found here: https://skriptlang.github.io/Skript
[19:46:29 INFO]: [Skript] Server Version: git-Paper-400 (MC: 1.17.1)
[19:46:29 INFO]: [Skript] Skript Version: 2.6
[19:46:29 INFO]: [Skript] Installed Skript Addons:
[19:46:29 INFO]: [Skript]  - skript-placeholders v1.5.2 (https://github.com/APickledWalrus/skript-placeholders)
[19:46:29 INFO]: [Skript]  - skript-gui v1.3-beta1 (https://github.com/APickledWalrus/skript-gui)
[19:46:29 INFO]: [Skript]  - Skellett v1.9.11 (https://forums.skunity.com/resources/skellett.24/)
[19:46:29 INFO]: [Skript]  - DiSky v3.6.0 (https://disky.itsthesky.info/)
[19:46:29 INFO]: [Skript]  - MorkazSk v1.2
[19:46:29 INFO]: [Skript]  - skRayFall v1.9.26 (https://sk.rayfall.net/)
[19:46:29 INFO]: [Skript]  - SkQuery v4.1.4
[19:46:29 INFO]: [Skript]  - SkBee v1.11.2 (https://github.com/ShaneBeee/SkBee)
[19:46:29 INFO]: [Skript]  - SkJade v1.3.2 (https://www.github.com/Ankoki-Dev/SkJade)
```

### Bug Description

When I ran the effect command `!send "%7 hours before {date}%"`, it sent the time 7 hours before {date}, but upon running it again, I noticed that it was decreasing by 7 hours every time I ran the same effect command. It seems that the expression `%timespan% before %date%` is broken with variables, as it shouldn't decrease the date.

### Expected Behavior

It should have just sent 7 hours before the date from the variable, which it did. But it also subtracted 7 hours from the date. So if it was 1/17/22 7:00PM, and you ran the effect command `!send "%7 hours before {date}%"`, it would send 1/17/22 12:00PM AND set the variable to 1/17/22 12:00PM.

### Steps to Reproduce

1. Enable effect commands thru config.sk (Haven't tested it without effect commands yet)
2. Run the command: `!set {date} to now`
3. Run the command: `!send "%7 hours before {date}%"` | Should send 7 hours before whatever %now% is.
4. Run the command: `!send "%7 hours before {date}%"` OR `!send "%{date}%"`

You should notice a difference in time.

### Errors or Screenshots

![time bug](https://user-images.githubusercontent.com/67490606/149675651-76ff91f2-0ffc-44b8-b5a5-56fad901f6c6.png)

### Other

IMPORTANT: I have not tested this thru normal code, only with effect commands, but I imagine it works the same.

### Agreement

- [X] I have read the guidelines above and confirm I am following them with this report.

**Comment by APickledWalrus** [comment]
Looking at the source expression, Skript should be cloning the date but isn't. Will be easy enough to fix though.

**Comment by AyhamAl-Ali** [comment]
Already fixed see #4432, you can wait until next release or use the nightly version

**Comment by APickledWalrus** [comment]
Duplicate of #4431
